gadgetapp/README.md

```diff
@@ -0,0 +1,118 @@
+            CHOICES = [(True, 'For True'), (False, 'For False')]
+            widgets = {
+                'pipeline_param_1': forms.TextInput(attrs={'class': 'config-input_field'}),
+                'pipeline_param_2': forms.NumberInput(attrs={'class': 'config-input_field'}),
+                'pipeline_param_3': forms.RadioSelect(choices=CHOICES),
+            }
+
+For this example, lets update pipeline configs to expose confidence level. To do that the models.py would be updated to this:
+
+    class UserPipelineConfig(models.Model):
+        confidence = models.FloatField(default=0.5)
+
+It's important that config name here matches the config name in the pipeline configs. If they match, any update to the config on the Gadget App will automatically be sent to the pipeline.
+
+Now the data structure has been updated, all that's left is to update the UI. The updated forms.py looks like this:
+
+    class ChangePipelineForm(forms.ModelForm):
+        class Meta:
+            model = UserPipelineConfig
+            fields = ('confidence')
+            labels = {'confidence': 'Confidence'}
+            widgets = {
+                'confidence': forms.NumberInput(attrs={'class': 'config-input_field'}),
+            }
+
+It's still referencing the same model so that doesn't change. The fields attribute needs to be updated to match the new config name. Labels, which gives the field a display name, needs to updated as well. Widgets is where the type of UI is defined. For this case the config is a float value so we choose NumberInput from the forms class.
+
+## Make Migrations
+
+Updating models.py won't have any effect unless the changes to the models are sent to the database. Django does this using migration files. At start up, the Gadget App runs *migrations* which goes through the migration files and makes sure the database schema is up to date. To add a new migration, run *makemigrations*. This can be done manually using the django manage functionality or the gadget app includes a docker-compose file that will take care of it. Simply run the command:
+
+    docker-compose -f docker-compose.makemigrations.yaml --env-file ../.env up
+
+## Initialize the Database
+
+The Gadget App relies on information from in the database from the other services in the Gadget. If that information isn't there the Gadget App will run init_db.py to initialize the database. It will need to create instances of sensor, pipeline, and automation configs for each instance of each of those services in the system. The instance names must also match the instance names of the services. It wil also need create instances of runtime status for each of the services. By default, it is setup to initialize a single sensor/pipeline gadget. Using a Gocator and with the pipeline and automation service name set to *gadget-pipeline* and *gadget-automation*. If those names change or there are more sensors/pipelines added to the system those will have to be added.
```

Comment:
``The Gadget App relies on information from in the database from the other services in the Gadget.``
Do you mean: ``from the database and from the other services in the Gadget``